src/collections.rs

```diff
@@ -173,46 +246,40 @@ pub unsafe extern "C" fn z_view_slice_from_str(
 #[no_mangle]
 #[allow(clippy::missing_safety_doc)]
 pub unsafe extern "C" fn z_view_slice_wrap(
-    this: *mut MaybeUninit<z_view_slice_t>,
+    this: &mut MaybeUninit<z_view_slice_t>,
     start: *const u8,
     len: usize,
 ) -> z_error_t {
-    if len == 0 {
-        z_view_slice_empty(this);
-        errors::Z_OK
-    } else if start.is_null() {
-        z_view_slice_empty(this);
-        errors::Z_EINVAL
-    } else {
-        Inplace::init(
-            this.transmute_uninit_ptr(),
-            CSlice::new_borrowed(start, len),
-        );
-        errors::Z_OK
+    match CSliceView::new(start, len) {
+        Ok(slice) => {
+            this.as_rust_type_mut_uninit().write(slice);
+            errors::Z_OK
+        }
+        Err(e) => e,
```

Comment:
we still need to initialize view slice to some defined value (i.e. empty slice), even in case of error

Reply:
done